05.wc/wc.rb

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'pathname'
+require 'find'
+
+def check_opt
+  opt = OptionParser.new
+  options = {}
+  opt.on('-l') { |lines| options[:l] = lines }
+  opt.on('-w') { |words| options[:w] = words }
+  opt.on('-c') { |byte| options[:c] = byte }
+  opt.parse!(ARGV)
+  options = { c: true, l: true, w: true } if options.empty?
+  options
+end
+
+def build_counts(str)
+  {
+    lines: str.count("\n"),
+    words: str.split(/\s+/).size,
+    characters: str.encode('UTF-8').bytesize
+  }
+end
+
+def display_totals(counts, opt)
+  print total_count(counts, :lines).to_s.rjust(8) if opt[:l]
+  print total_count(counts, :words).to_s.rjust(8) if opt[:w]
+  print total_count(counts, :characters).to_s.rjust(8) if opt[:c]
+  puts ' total'
+end
+
+def total_count(counts, key)
+  counts.sum { |count| count[key] }
+end
+
+def display_counts(counts, opt)
+  counts.each_with_index do |count, index|
+    print count[:lines].to_s.rjust(8) if opt[:l]
+    print count[:words].to_s.rjust(8) if opt[:w]
+    print count[:characters].to_s.rjust(8) if opt[:c]
+    puts " #{ARGV[index] unless ARGV.empty?}"
+  end
+  return if ARGV.empty?
+
+  display_totals(counts, opt)
+end
+
+def check_inputs
+  opt = check_opt
+  counts = if ARGV.empty?
```

Comment:
こちらもMUSTではないですが、

```suggestion
  counts =
    if ARGV.empty?
```

のようにifの手前で改行するのが個人的には好きだったりします。
https://qiita.com/jnchito/items/9b3f476f40803444b471

Reply:
なるほど、見た目だけでなく利点があるのですね。
手前で改行するように変更しました。